File: relative-ranks/relative-ranks.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        
        int n = score.length;
        
        int rS[][] = new int[n][2];
        
        String ranks[] = new String[n];
        
        for (int i = 0; i < n; ++i) {
            
            rS[i][0] = score[i];
            rS[i][1] = i;
        }
        
        Arrays.sort(rS, (a, b) -> (b[0] - a[0]));
        
        for (int j = 0; j < n; ++j) {
            
            int place = rS[j][1];
            
            ranks[place] = "" + (j + 1);
            
            if (j == 0) { ranks[place] = "Gold Medal"; }
            if (j == 1) { ranks[place] = "Silver Medal"; }
            if (j == 2) { ranks[place] = "Bronze Medal"; }
        }

        return ranks;
    }
}

File: relative-ranks/relative-ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        
        vector<string> rank(score.size(), "");
        
        priority_queue<pair<int, int>> pQ;
        
        for (int i = 0; i < score.size(); ++i) {
            
            pQ.push(make_pair(score[i],i));
        }
        
        for (int j = 0; j < rank.size(); ++j) {
            
            pair<int,int> r = pQ.top();
            pQ.pop();
            
            rank[r.second] = to_string(j + 1);
            
            if (j == 0) { rank[r.second] = "Gold Medal"; }
            if (j == 1) { rank[r.second] = "Silver Medal"; }
            if (j == 2) { rank[r.second] = "Bronze Medal"; }
            
        }
        
        return rank;
    }
};